File: java/src/main/java/gov/doc/ntia/sigmf/ext/annotation/algorithm/TimeDomainDetection.java
package gov.doc.ntia.sigmf.ext.annotation.algorithm;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@Deprecated
@JsonInclude(JsonInclude.Include.NON_NULL)
public class TimeDomainDetection extends DomainDetection {

  private static final long serialVersionUID = 1L;

  @JsonProperty(value = "ntia-algorithm:number_of_samples", required = true)
  protected Long numberOfSamples;

  public Long getNumberOfSamples() {
    return numberOfSamples;
  }

  public void setNumberOfSamples(Long numberOfSamples) {
    this.numberOfSamples = numberOfSamples;
  }
}

File: Csharp/sigmf/sigmf/ext/Annotationdir/algorithm/TimeDomainDetection.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    [Serializable()]
    public class TimeDomainDetection : DomainDetection
    {
        private static readonly long serialVersionUID = 1L;

        [JsonProperty("ntia-algorithm:number_of_samples")]
        protected long numberOfSamples;

        public long NumberOfSamples { get => numberOfSamples; set => numberOfSamples = value; }

        public TimeDomainDetection() {
            this.detectionDomain = "time";
        }
    }
}
